Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/ErrorHandler.java

```diff
@@ -114,27 +119,31 @@ public ErrorHandler(int errorHorizon, int forecastHorizon, int sequenceIndex, do
         int currentLength = (actualsFlattened == null) ? 0 : actualsFlattened.length;
         checkArgument(currentLength % inputLength == 0, "actuals array is incorrect");
         int forecastLength = (pastForecastsFlattened == null) ? 0 : pastForecastsFlattened.length;
-        checkArgument(forecastLength == currentLength * dimensions * 3 / inputLength, "misaligned forecasts");
+
         int arrayLength = max(forecastHorizon + errorHorizon, currentLength / inputLength);
         this.pastForecasts = new RangeVector[arrayLength];
         this.actuals = new float[arrayLength][inputLength];
 
         int length = forecastHorizon * inputLength;
+        checkArgument(forecastLength == currentLength * dimensions * 3 * forecastHorizon / inputLength,
+                "misaligned forecasts");
 
         this.errorMean = new float[length];
         this.errorRMSE = new DiVector(length);
         this.intervalPrecision = new float[length];
+        this.adders = new RangeVector(length);
         this.multipliers = new RangeVector(length);
         this.errorDistribution = new RangeVector(length);
 
+        int pastForecastsLength = forecastHorizon * dimensions;
```

Comment:
pastForecastsLength = forecastHorizon * inputLength ? Carrying over the change from the previous comment. Dimensions are being used in the underlying RCF and can be confusing.

Reply:
yes, changed